java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -152,12 +152,22 @@ ExecutionPath addEdge(ExplodedGraph.Edge edge) {
       Symbol newTrackSymbol = newTrackedSymbol(edge);
 
       Set<LearnedConstraint> learnedConstraints = learnedConstraints(edge);
-      List<JavaFileScannerContext.Location> lcFlow = learnedConstraints.stream()
+      List<JavaFileScannerContext.Location> lcFlow = flowFromLearnedConstraints(edge, learnedConstraints);
+      flowBuilder.addAll(lcFlow);
+
+      if (isMethodInvocationNode(edge.parent)) {
```

Comment:
I would have this test within the flow from yield method in fact, so the algorithm is pretty straightforward and the method to compute flow from yield will only do so if required... (and return an emtpy list otherwise). (In fact if I'm correct you actually don't need that test at all...)